iotedgedev/cli.py

```diff
@@ -304,7 +304,7 @@ def stop_simulator():
               help="Specify number of milliseconds to monitor for messages")
 def monitor(timeout):
     utility = Utility(envvars, output)
-    ih = IoTHub(envvars, utility, output)
+    ih = IoTHub(envvars, utility, output, azure_cli)
     ih.monitor_events(timeout)
```

Comment:
What's the unit of `timeout`? Is it still milliseconds? The behavior seems to be inconsistent with Python 2 and Python 3.6. With Python 3.6, I have specified 10 as timeout, but the monitor session ran several minutes.

Reply:
Both should be seconds:
https://github.com/Azure/iothub-explorer/blob/master/iothub-explorer-monitor-events.js#L45

https://github.com/Azure/azure-iot-cli-extension/blob/e0984b1b132e296962a92a0637a3fb59c6d2d0f1/azext_iot/operations/hub.py#L1125

https://docs.microsoft.com/en-us/cli/azure/ext/azure-cli-iot-ext/iot/hub?view=azure-cli-latest#ext-azure-cli-iot-ext-az-iot-hub-monitor-events

@digimaun - Is this correct?

Reply:
I think I found out the cause of the inconsistency. The `timeout` option for `iothub-explorer monitor-events` specifies "Exit after the given number of seconds". In comparison, the `timeout` option for `az iot hub monitor-events` specifies "Maximum seconds to maintain connection __without receiving message__". Since my Edge device is continuously sending messages to IoT Hub, when monitoring on Python 3.6, the monitor session will never stop.

Reply:
Yes the extension timeout is in seconds and @LazarusX's comment is right on the timeout behavior for monitor-events

Reply:
@digimaun - I'm trying to add a timeout to process.communicate to force kill the monitor-events process.  Can you review this commit and LMK why you think it is not actually killing the process.  Also, can you recommend other ways to get this behavior?  Thanks!

https://github.com/jongio/iotedgedev/commit/ab80d0d724e0ff9f5c0d4c9f1ebc752e71f90dbd

Reply:
I sent a PR to your personal repo on the same processtimeout branch.

Reply:
Thx.  You are a master!

Reply:
@jongio @digimaun Do you plan to merge the `processtimeout` branch to `Azure/iotedgedev:master` branch?

Reply:
It didn’t fully work yet. paymaun will have a look this week to figure out why tod is failing. Please feel free to try if you have time. You can pull fork